### PR TITLE
E2E Tests: Migrate to github actions

### PR DESCRIPTION
**Pull request by brbrr**
Move E2E tests from Travis into GH actions

With GH Actions, we could customize our runs, run different suites on different events (e.g. on new tags, merges to master, etc)

#### Jetpack product discussion

n/a 
#### Does this pull request change what data or activity we track or use?

n/a 
#### Testing instructions:

* Take a look at "Checks" section
* Make sure the tests have passed.

#### Proposed changelog entry for your changes:

* n/a

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -81,41 +81,41 @@ matrix:
...
-    env: LATEST_GUTENBERG=1
-    if: branch !~ /(^branch-.*-built)/ AND env(RUN_E2E) = true AND fork = false
-    before_script:
-    - nvm install
-    - yarn build-production
-    - ./tests/e2e/bin/env.sh start
-    - yarn test-decrypt-config
-    script:
-    - yarn test-e2e --testPathIgnorePatterns=updater
+  # - name: "E2E tests"
```
Comment (jeherve) [inline .travis.yml:84]: Maybe we can remove it all instead of commenting it out?

File: .github/workflows/e2e-tests.yml [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    name: All
+    runs-on: ubuntu-latest
+
+    steps:
+
+    - run: |
+        echo $GITHUB_RUN_NUMBER
+        echo $GITHUB_RUN_ID
+        echo $GITHUB_ACTION
+        cat $GITHUB_EVENT_PATH
```
Comment (jeherve) [inline .github/workflows/e2e-tests.yml:23]: Is this needed or was this for debugging only?

**Review by jeherve — COMMENTED** [review]
The tests seem to pass.

I was a bit worried about potentially having secrets slip out in the builds, but couldn't find anything bad.

I only have 2 questions below.

**Comment by kraftbj** [comment]
I really dig the approach here. The action is erroring, though, `ERROR: WP_SITE_URL is not set after launching an ngrok`.

I think it'll give us a lot of future flexibility, so I'm good to roll through it once it passes.

**Comment by fullofcaffeine** [comment]
Hi @brbrr 👋🏻 

We're currently [working](https://href.li/?https://github.com/Automattic/wp-calypso/issues/46583) to better understand what would it take to have E2E tests in other repos use a specific/latest version of Gutenberg*. @ockham pointed me to the work you've done [here](https://href.li/?https://github.com/Automattic/jetpack/pull/15784), and also mentioned this PR as we also want to understand what (mostly E2E) tooling/infrastructure is available for each project, and Jetpack seems to have have the edge for now as a template/inspiration we could use for other projects.

We also would like to wait for this PR to be merged before adding any new tests (as part of this other related [task](https://href.li/?https://github.com/Automattic/wp-calypso/issues/46584)).

That being said, how can we help you testing this PR? I see the testing instructions just mention checking if all tests pass in the list of GH jobs below, as some are failing. Is there anything we can do to help you?

Thanks! 🙇🏻 

*See pc9Uba-8b-p2

**Review by kraftbj — APPROVED** [review]
This looks good to me. Added a minor commit to remove the allowed-failure in travis for the e2e-guten test.

**Comment by brbrr** [comment]
Hey @fullofcaffeine! Sorry for the late response.

The PR is merged, so nothing to test here :) There is this PR in the works (it is ready for review, but I still want to double-check it if works fine) https://github.com/Automattic/jetpack/pull/17523

It is infrastructure related, but with that PR merged, we would be able to run multiple E2E containers in parallel. I would appreciate a review :)

**Pull request merged**